src/DotNetCore.CAP.NATS/ITransport.NATS.cs

```diff
@@ -31,14 +32,10 @@ public Task<OperateResult> SendAsync(TransportMessage message)
 
             try
             {
-                var binFormatter = new BinaryFormatter();
-                using var mStream = new MemoryStream();
-                binFormatter.Serialize(mStream, message);
-
-                //connection.Publish(message.GetName(), mStream.ToArray());
-                //return Task.FromResult(OperateResult.Success);
-
-                var reply = connection.Request(message.GetName(), mStream.ToArray(), 2000);
+                var json= JsonSerializer.Serialize(message);
```

Comment:
We have the serialization interface,  need to use `ISerializer` instead here

Reply:
Do you mean DotNetCore.CAP.Serialization.ISerializer?

Reply:
I have checked the DotNetCore.CAP.Serialization.ISerializer and JsonUtf8Serializer, they are based on Message not TransportMessage, may I add another method for TransportMessage?

Reply:
Consider using the version `0.14.0-pre1`, which provides support for headers

https://github.com/nats-io/nats.net/releases/tag/0.14.0-pre1

Reply:
Thanks, I have updated to `0.14.0-pre1` now, and the interface `ISerializer`  is no longer need, so I removed it.